api/defaults.go

```diff
@@ -0,0 +1,819 @@
+		Enabled: true,
+		URL:     "https://mainnet.infura.io/v3/800c641949d64d768a5070a1b0511938",
+	}
+
+	nodeConfig.Name = "StatusIM"
+	nodeConfig.Rendezvous = false
+	clusterConfig, err := params.LoadClusterConfigFromFleet("status.prod")
+	if err != nil {
+		return nil, err
+	}
+	nodeConfig.ClusterConfig = *clusterConfig
+	nodeConfig.NoDiscovery = true
+	nodeConfig.MaxPeers = 20
+	nodeConfig.MaxPendingPeers = 20
+
+	nodeConfig.WalletConfig = params.WalletConfig{Enabled: true}
+	nodeConfig.LocalNotificationsConfig = params.LocalNotificationsConfig{Enabled: true}
+	nodeConfig.BrowsersConfig = params.BrowsersConfig{Enabled: true}
+	nodeConfig.PermissionsConfig = params.PermissionsConfig{Enabled: true}
+	nodeConfig.MailserversConfig = params.MailserversConfig{Enabled: true}
+	nodeConfig.EnableNTPSync = true
+
+	nodes := []string{"enrtree://AOGECG2SPND25EEFMAJ5WF3KSGJNSGV356DSTL2YVLLZWIV6SAYBM@prod.nodes.status.im"}
+	nodeConfig.ClusterConfig.WakuNodes = nodes
+	nodeConfig.ClusterConfig.DiscV5BootstrapNodes = nodes
+
+	nodeConfig.WakuV2Config = params.WakuV2Config{
+		Enabled:        true,
+		EnableDiscV5:   true,
+		DiscoveryLimit: 20,
+		Host:           "0.0.0.0",
+		AutoUpdate:     true,
+		PeerExchange:   true,
+	}
+
+	if request.WakuV2Nameserver != nil {
+		nodeConfig.WakuV2Config.Nameserver = *request.WakuV2Nameserver
+	}
+
+	nodeConfig.ShhextConfig = params.ShhextConfig{
+		BackupDisabledDataDir:      request.BackupDisabledDataDir,
+		InstallationID:             installationID,
+		MaxMessageDeliveryAttempts: 6,
+		MailServerConfirmations:    true,
+		VerifyTransactionURL:       "",
+		VerifyENSURL:               "",
+		VerifyENSContractAddress:   "",
+		VerifyTransactionChainID:   1,
+		DataSyncEnabled:            true,
+		PFSEnabled:                 true,
+	}
+
+	if request.VerifyTransactionURL != nil {
+		nodeConfig.ShhextConfig.VerifyTransactionURL = *request.VerifyTransactionURL
+	}
+
+	if request.VerifyENSURL != nil {
+		nodeConfig.ShhextConfig.VerifyENSURL = *request.VerifyENSURL
+	}
+
+	if request.VerifyTransactionChainID != nil {
+		nodeConfig.ShhextConfig.VerifyTransactionChainID = *request.VerifyTransactionChainID
+	}
+
+	if request.VerifyENSContractAddress != nil {
+		nodeConfig.ShhextConfig.VerifyENSContractAddress = *request.VerifyENSContractAddress
+	}
+
+	if request.LogLevel != nil {
+		nodeConfig.LogLevel = *request.LogLevel
+		nodeConfig.LogEnabled = true
+
+	} else {
+		nodeConfig.LogEnabled = false
+	}
+
+	return nodeConfig, nil
+}
+
+func buildSigningPhrase() (string, error) {
+	length := big.NewInt(int64(len(dictionary)))
+	a, err := rand.Int(rand.Reader, length)
+	if err != nil {
+		return "", err
+	}
+	b, err := rand.Int(rand.Reader, length)
+	if err != nil {
+		return "", err
+	}
+	c, err := rand.Int(rand.Reader, length)
+	if err != nil {
+		return "", err
+	}
+
+	return dictionary[a.Int64()] + " " + dictionary[b.Int64()] + " " + dictionary[c.Int64()], nil
+
+}
+
+var dictionary = []string{
```

Comment:
@ilmotta / @cammellos Sorry I'm even more late to this discussion. Can I ask are we wanting to overwrite select fields of a default configuration with a some input configuration?

Like this https://go.dev/play/p/fSRVAnr2F7K ?

```go
type Cthulhu struct {
    AeonsSlumbering      uint
    SummoningIncantation string
    Awoken               bool
    PlanesOccupied       int
    Sacrifices           []string
}

func TestJSONMarshalling(t *testing.T) {
    defaultCthulhu := &Cthulhu{
        AeonsSlumbering:      6,
        SummoningIncantation: "ph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn",
        Awoken:               false,
        PlanesOccupied:       -3,
        Sacrifices:           []string{"Sam", "Andrea", "Vitaliy"},
    }

    incomingJSON := `{"Awoken":true,"Sacrifices":["Our Dignity"]}`

    err := json.Unmarshal([]byte(incomingJSON), defaultCthulhu)
    if err != nil {
        fmt.Println(err.Error())
    }
    fmt.Printf("%+v\n", defaultCthulhu)
}
```

```text
=== RUN   TestJSONMarshalling
&{AeonsSlumbering:6 SummoningIncantation:ph'nglui mglw'nafh Cthulhu R'lyeh wgah'nagl fhtagn Awoken:true PlanesOccupied:-3 Sacrifices:[Our Dignity]}
--- PASS: TestJSONMarshalling (0.00s)
PASS
```

Reply:
Yes @Samyoul, something like that was in my mind (although with something like TOML that supports comments). I understand increasing configurability via external mechanisms is sometimes hard to justify, since the hard coded alternative works well.

In the end, I don't know if separating configuration from code will bring enough value versus the effort + convenience of just changing the code. It's just an idea to be considered in future iterations.

I found that sometimes an external configuration forced me to create systems that made less assumptions, since the config state could be more easily changed by consumers (devs, services, etc). Configurability is usually brought up in discussions related to 12-factor apps or when there's an intention of applying some of the principles, but perhaps this is a weak argument for status-go, right?

Reply:
I think external config is fine, for overriding the JSON, in theory works ok, and that was our initial idea with @Samyoul , but then you look at what we need to pass, and it's all deeply nested JSON, so it looks like we are bleeding out a lot of implementation details to the client.
For example, say the client wants to override the fleet for waku2.
The will have to pass JSON that looks like:

`{NodeConfig: {WakuV2Config: {Enabled: true, Fleet: "...."}}, RootDirectory: "..."}`

and even if we make sure that only relevant parts are overriden, so it doesn't have to pass a full map if only overriding a single value, I can't think that's better than a flat structure:

`{:WakuV2Fleet : "status.prod", :rootDirectory ...}`

So we are not exposing the structure of our internal config to the client, but still allow to customize some (not all, that's the drawback I guess) fields

Reply:
>So we are not exposing the structure of our internal config to the client, but still allow to customize some (not all, that's the drawback I guess) fields

Good point, we can create an interface to clients that's more stable than our internal representation.

My only gripe with JSON is that it is kind of a bad media for human consumption, and IMO should be generally relegated for system-to-system communication. Something like a git config file (INI) or [TOML](https://toml.io/en/) would be nice, since the file can be nicely documented for humans.